homeassistant/components/light/abode.py

```diff
@@ -0,0 +1,96 @@
+"""
+This component provides HA light support for Abode Security System.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/light.abode/
+"""
+import logging
+
+from homeassistant.components.abode import AbodeDevice, DOMAIN
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, ATTR_RGB_COLOR,
+    SUPPORT_BRIGHTNESS, SUPPORT_RGB_COLOR, Light)
+
+
+DEPENDENCIES = ['abode']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up Abode light devices."""
+    import abodepy.helpers.constants as CONST
+
+    data = hass.data[DOMAIN]
+
+    device_types = [CONST.TYPE_LIGHT, CONST.TYPE_SWITCH]
+
+    devices = []
+
+    # Get all regular lights that are not excluded or switches marked as lights
+    for device in data.abode.get_devices(generic_type=device_types):
+        if device.device_id in data.exclude:
+            continue
+
+        if (device.generic_type == CONST.TYPE_SWITCH and
```

Comment:
It would make sense to add a helper method on `hass.data[DOMAIN]` that given a device type, will give you a list that takes exclusion into account. That way you are not repeating yourself in every platform.